core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStackFactory.java

```diff
@@ -61,7 +62,7 @@ protected void setTextProvider(TextProvider textProvider) {
         this.textProvider = textProvider;
     }
     
-    @Inject(value="allowStaticMethodAccess", required=false)
+    @Inject(value = XWorkConstants.ALLOW_STATIC_METHOD_ACCESS, required = false)
```

Comment:
👏

Reply:
Hello @lukaszlenart .

The only other consideration might be whether to add in the changes being reviewed for SecurityMemberAccess from PR#320 (the elements valid for 2.5.x anyway ... minus the new field access flag), provided they make sense for 2.5.x.

If you think those changes (the ordering optimization for public state checks, et. al.) would be valid here as well I can make a single commit update for SecurityMemberAccess to this PR.

Please let me know what you think.  :)

Reply:
This PR is already huge with a long discussion, let's merge it first then we can work on another issue.